honeybee/aperturetype.py

```diff
@@ -34,18 +32,33 @@ class Window(_ApertureType):
     pass
 
 
-class _Types(object):
-    """Aperture types."""
+class _ApertureTypes(object):
+    """Aperture types enumeration."""
+
+    _window = Window()
 
     def __init__(self):
-        self._window = Window()
+        pass
 
     @property
     def window(self):
         return self._window
 
+    def by_name(self, aperture_type_name):
+        """Get an Aperture Type instance from its name.
+
+        Args:
+            aperture_type_name: A text string for the aperture type (eg. "Window").
+        """
+        attr_name = re.sub('(?<!^)(?=[A-Z])', '_', aperture_type_name).lower()
+        try:
+            return getattr(self, attr_name)
+        except AttributeError:
+            raise AttributeError('Aperture Type "{}" is not supported by this '
+                                 'installation of honeybee.'.format(aperture_type_name))
```

Comment:
The primary case where I was thinking someone would get this error is when they have created a model using an installation of honeybee with honeybee-energy installed so they assign `GlassDoor` to an Aperture. Then, an installation of honeybee-core without the energy extension goes to re-serialize it and it can't because there's no `GlassDoor` in honeybee-core.

So this is why I thought it was more pertinent to note the installation of honeybee in the message.  Let me know if you still feel we should change it now that I have told you the main case where I expect it to be raised.

Reply:
I see. We should probably more explicit about this and just include parts of what you said. Something like to support this case ensure that `honeybee-energy` extension is installed.

Reply:
Hm. Well there isn't really a way to tell if the aperture type was added by `honeybee-energy` or another plugin.  I could add something like this:

```
Aperture Type "{}" is not supported by honeybee. Make sure that you have all necessary honeybee extensions installed'.format(aperture_type_name)
```

Although, the more that I think about it, the more it feels like we are unnecessarily limiting the ability for models to be serialized into a Radiance format just because there is an aperture type that doesn't have much meaning for honeybee-radiance.  It really seems like a shame that we will be unable to set up cloud services with just honeybee-radiance that can re-serialize a model from our plugins just because someone put a glass door in there.  Maybe we really should consider including the other 2 aperture types in the core.

Reply:
I guess the other option is to just have the type default to `Window` if the type is not recognized.  But it also feels strange to me that you wouldn't want to have this information about whether the Aperture is a Glass Door in honeybee-radiance.  Are you not planning on having something like a `MaterialSet` in honeybee-radiance, which can auto-assign radiance materials depending on face type or aperture type?

Reply:
I actually just implemented `MaterialSet` which should probably help with this. We can leave it for now and re-address it if it was an issue in the future.

Reply:
Glad to hear that you implemented `MaterialSet`.  That is going to be particularly helpful for making large daylight models and adding urban daylight capabilities to Dragonfly.  And, ok, we can leave everything as it is for now but I have a strong intuition that we will want to move all of the aperture and face types to the core at some point.